30413/Paltineanu Rares - Mihai/Assignment1/main.c

```diff
@@ -0,0 +1,331 @@
+
+    }
+
+    if(firstOrLast == 1) { ///first
+
+        arrayT* current = first;
+        first = first->next;
+        free(current);
+
+    } else { ///last
+
+        arrayT* current = first;
+
+        while(current->next != last)
+            current = current->next;
+
+        last = current;
+        current = last->next;
+        free(current);
+        last->next = NULL;
+
+    }
+
+}
+
+int takingTheNumber(char* array) {
+
+    int i = 0;
+    int number = 0;
+    int negativenumber = 0;
+
+    while(!(array[i] >= '0' && array[i] <= '9'))
+        i++;
+
+    if(array[i - 1] == '-') {
+
+        negativenumber = -1;
+
+    } else {
+
+        negativenumber = 1;
+
+    }
+
+    while(array[i] >= '0' && array[i] <= '9' && array[i] != '\0') { /// add first
+
+        number = number * 10 + (array[i] - '0');
+        i++;
+
+    }
+
+    number*= negativenumber;
+
+    return number;
+}
+
+void printFirstX(int number) {
+
+    arrayT* current = first;
+    int i = 0;
+
+    while(i < number) {
+
+        printf("%d ", current->element);
+        current = current->next;
+        i++;
+
+    }
+    printf("\n");
+
+}
+
+void printLastX(int number) {
+
+    if(first == NULL)
+        return;
+
+    arrayT* current = first;
+    int i = 0;
+
+    while(i < ct - number) {
+
+        current = current->next;
+        i++;
+    }
+
+    while(current != NULL) {
+
+        printf("%d ", current->element);
+        current = current->next;
+
+    }
+
+    printf("\n");
+
+}
+
+void printall() {
+
+
+    if(first == NULL) {
+
+        printf("There are no elements in the list.\n");
+
+    } else {
+
+
+        arrayT* current = first;
+
+        while(current != NULL) {
+
+            printf("%d ", current->element);
+            current = current->next;
+
+        }
+    }
+
+    printf("\n");
+
+}
+
+void removeX(int number) {
+
+    if(first != NULL) {
+
+        arrayT* previous = (arrayT*)malloc(sizeof(arrayT));
+        arrayT* current = first;
+
+        while(current != NULL) {
+
+            if(current->element == number) {
+
+                ct--;
+
+                if(current == first) {
+
+                    first = first->next;
+
+                } else if(current == last) {
+
+                    last = previous;
+                    last->next = NULL;
+
+                } else {
+
+                    previous->next = current->next;
+
+                }
+
+                previous = current;
+                current = current->next;
+                free(previous);
+                previous = current;
+
+            } else {
+
+                previous = current;
+                current = current->next;
+
+            }
+        }
+    }
+}
+
+void readingCommand(FILE* in) {
+
+    char* input = (char*)malloc(50);
+    char* command = (char*)malloc(30);
+
+    int number;
+
+    while(fgets(input, 50, in) != NULL) {
+
+        command = takingTheCommand(input);
+
+        if(stricmp(command, command1) == 0) { ///first
```

Comment:
This is also great - defining constants and comparing the read strings with them, but then why not name the constants properly?
```
#define ADD_FIRST "AF"
```

for example - then you would have

`if(stricmp(command, ADD_FIRST) == 0)`

and suddenly, everyone reading your code will understand what is going on 😄

Reply:
Actually, i don't know why i didn't name the constants properly, i will do it next time